catalogs/usfimr-s1/request_builders.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python
+
+import urllib3
+import datetime
+import time
+import json
+import sys
+
+
+def get_sentinelhub_token(oauth_id, oauth_secret, connection_pool):
+    """ Example curl:
+    curl --request POST --url https://services.sentinel-hub.com/oauth/token
+            -H "content-type: application/x-www-form-urlencoded"
+            -d 'grant_type=client_credentials&client_id=<client_id>'
+            --data-urlencode 'client_secret=<client_secret>'
+    """
+    r = connection_pool.request(
+        "POST",
+        "/oauth/token",
+        encode_multipart=False,
+        headers={"Content-Type": "application/x-www-form-urlencoded"},
+        fields={
+            "grant_type": "client_credentials",
+            "client_id": oauth_id,
+            "client_secret": oauth_secret,
+        },
+    )
+    if r.status == 200:
+        one_hour_later = datetime.datetime.now() + datetime.timedelta(hours=1)
+        print(
+            "Successfully acquired Sentinel-Hub token. It will be valid until {}".format(
+                one_hour_later
+            )
+        )
+        return json.loads(r.data.decode("utf-8"))["access_token"]
+    else:
+        sys.exit("Failure to acquire Sentine-Hub token. Status: {}".format(r.status))
+
+
+def search_sentinelhub_s1(flood_temporal_bounds, flood_bbox, connection_pool, token):
+    """ Example curl:
+        curl -X "POST" "https://services.sentinel-hub.com/api/v1/catalog/search" \
+            -H 'Content-Type: application/json' \
+            -H 'Authorization: Bearer <token>' \
+            -d $'{
+                "limit": 1,
+                "fields": {
+                    "include": [
+                        "properties.eo:gsd"
+                    ]
+                },
+                "datetime": "2019-05-22T00:00:00Z/2019-05-23T00:00:00Z",
+                "collections": [
+                    "sentinel-1-grd"
+                ],
+                "bbox": [
+                    -94.3670654296875,
+                    38.371808917147554,
+                    -95.712890625,
+                    40.30885442563764
+                ]
+            }'
+    """
+    parameters = {
+        "fields": {"include": ["properties.eo:gsd"]},
+        "datetime": flood_temporal_bounds,
+        "collections": ["sentinel-1-grd"],
+        "bbox": flood_bbox,
+    }
+    encoded = json.dumps(parameters).encode("utf-8")
+    r = connection_pool.request(
+        "POST",
+        "/api/v1/catalog/search",
+        headers={
+            "Content-Type": "application/json",
+            "Authorization": "Bearer {}".format(token),
+        },
+        body=encoded,
+    )
+    return r
+
+
+def create_batch_request(flood_item, flood_temporal_bounds, connection_pool, token):
```

Comment:
I realize we likely don't reuse this, but the method arguments seem overly specific for some of the methods in this file. 

In this particular case, perhaps swapping `flood_item` and `flood_temporal_bounds` to:
- `defaultTilePath`
- `geometry`
- `start_time`
- `end_time`

Would be both clearer for the person calling this method (i.e. its unclear what the type and shape of `flood_temporal_bounds` is without digging into the implementation) and easier to make generic later.